tests/gamma/csm_observability_test.py

```diff
@@ -45,39 +100,389 @@ def setUpClass(cls):
+                self.assertEqual("prometheus_target", series.resource.type)
+                for label_key, label_value in series.resource.labels.items():
+                    if label_key == "project_id":
+                        self.assertEqual(self.project, label_value)
+                    elif label_key == "namespace" and self._is_client_metric(
+                        metric
+                    ):
+                        # client namespace
+                        self.assertEqual(self.client_namespace, label_value)
+                    elif label_key == "namespace" and self._is_server_metric(
+                        metric
+                    ):
+                        # server namespace
+                        self.assertEqual(self.server_namespace, label_value)
+                    elif label_key == "job" and self._is_client_metric(metric):
+                        # the "job" label on the monitored resource refers to
+                        # the GMP PodMonitoring resource name
+                        self.assertEqual(
+                            self.client_runner.pod_monitoring_name,
+                            label_value,
+                        )
+                    elif label_key == "job" and self._is_server_metric(metric):
+                        # the "job" label on the monitored resource refers to
+                        # the GMP PodMonitoring resource name
+                        self.assertEqual(
+                            self.server_runner.pod_monitoring_name,
+                            label_value,
+                        )
+                    elif label_key == "cluster":
+                        self.assertIn(
+                            label_value, xds_k8s_flags.KUBE_CONTEXT.value
+                        )
+                    elif label_key == "instance" and self._is_client_metric(
+                        metric
+                    ):
+                        # the "instance" label on the monitored resource refers
+                        # to the GKE pod and port
+                        self.assertIn(test_client.hostname, label_value)
+                    elif label_key == "instance" and self._is_server_metric(
+                        metric
+                    ):
+                        # the "instance" label on the monitored resource refers
+                        # to the GKE pod and port
+                        self.assertIn(test_server.hostname, label_value)
+
+        # This tests whether each of the "byes sent" histogram type metric
+        # should have at least 1 data point whose mean should converge to be
+        # close to the number of bytes being sent by the RPCs.
+        with self.subTest("10_check_bytes_sent_vs_data_points"):
+            for metric in _CLIENT_SENT_METRICS:
+                self.assertTrue(
+                    self._at_least_one_point_within_range(
+                        all_results[metric][0].points, _REQUEST_PAYLOAD_SIZE
```

Comment:
note that with serialization, the actual value that you see would be slightly different, but I would expect to always see the same value.

Reply:
Yea I left a 5% margin in there, which should be good enough.